### PR TITLE
Toolbar fixes

### PR DESCRIPTION
**Pull request by euismod2336**
### Summary of Changes
Added "sections" for toolbar buttons. Removed the mod_toolbar due to the buttons being in different places now. Config buttons and breadcrumbs are now shown from the template directly, kept the toolbar module position for b/c. The other toolbar buttons are now loaded via the 'searchtools' layout files.

Removed the mod_toolbar from the sql/postgresql files so shouldn't be installed via the installation anymore.

@coolcat-creations Some css changes will have to be done to get the layout to be what you want (if you want any html output to change, let me know)

@roland-d Can you check if i've missed any install/update script?

### CODE REVIEW

**Comment by euismod2336** [comment]
Why when i merge the release branch with my branch, the install scripts don't come up as conflicts?

**Comment by coolcat-creations** [comment]
Thank you I check it out today!!! :) Regarding the conflicts i cc @bembelimen - because I don't know...

File: libraries/src/Toolbar/Toolbar.php [hunk 1]
```diff
@@ -53,12 +54,12 @@ class Toolbar
...
 
 	/**
-	 * Toolbar array
+	 * Toolbar array, we keep track of buttons in specific sections.
 	 *
 	 * @var    array
 	 * @since  1.5
 	 */
-	protected $_bar = [];
+	protected $_bar = array(array());
```
Comment (roland-d) [inline libraries/src/Toolbar/Toolbar.php:62]: @euismod2336 Why not use short array syntax here?

File: libraries/src/Toolbar/Toolbar.php [hunk 2]
```diff
@@ -374,8 +355,8 @@ public function render(array $options = [])
 	 *
 	 * @return  string
 	 *
+	 * @throws  Exception
```
Comment (roland-d) [inline libraries/src/Toolbar/Toolbar.php]: @euismod2336 In Joomla they want the `@throws` after the `@since` tag.

**Comment by roland-d** [comment]
@euismod2336 Good work !! I added some remarks. The only thing I see missing is the update SQL files where you delete the mod_toolbar from existing installations. These are the files in the folder `administrator/components/com_admin/sql/updates/` There create a new file with the current date and add the SQL statements that remove the database entry from old installations. `DELETE FROM ...` That should be it.

**Comment by coolcat-creations** [comment]
Hi there!

Not sure if my npm is working correct... It looks like this here:
grafik
and the breadcrumbs are sticky instead of the buttons below:
grafik
And the button design is completely broken...
Should look like this:
grafik
I don't know unfortunately how to include the custom elements there - can you help?

**Comment by euismod2336** [comment]
found the issue, hadn't updated the release branch .__. should be fine now

File: libraries/src/Toolbar/Toolbar.php (hunk 1 above)
Comment (euismod2336) [reply libraries/src/Toolbar/Toolbar.php:62]: can't init array in array with short syntax, maybe array([]) would work but that seems less readable

**Comment by euismod2336** [comment]
@coolcat-creations yes, like i said; there need to be some css changes as the buttons have a new location and the css that makes the breadcrumbs sticky needs to be set on the "toolbar" buttons.

File: libraries/src/Toolbar/Toolbar.php (hunk 1 above)
Comment (roland-d) [reply libraries/src/Toolbar/Toolbar.php:62]: I kinda figured but wanted to be sure. In that case, keep it as it is.

**Comment by euismod2336** [comment]
Added 'delete' statements in the update scripts in sql & postgresql

File: administrator/components/com_admin/sql/updates/postgresql/4.0.0-2019-03-31.sql [hunk 3]
```diff
@@ -1,3 +1,5 @@
 INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder", "client_id", "enabled", "access", "protected", "manifest_cache", "params", "checked_out", "checked_out_time", "ordering", "state") VALUES
 (0, 'plg_sampledata_multilang', 'plugin', 'multilang', 'sampledata', 0, 1, 1, 0, '', '{}',  0, '1970-01-01 00:00:00', 0, 0),
 (0, 'plg_installer_webinstaller', 'plugin', 'webinstaller', 'installer', 0, 1, 1, 0, '', '{}',  0, '1970-01-01 00:00:00', 0, 0);
+DELETE FROM `#__extensions` WHERE `name` = 'mod_toolbar';
```
Comment (roland-d) [inline administrator/components/com_admin/sql/updates/postgresql/4.0.0-2019-03-31.sql]: @euismod2336 The backticks must be double-quotes in Postgres.

**Comment by coolcat-creations** [comment]
Ok, just the buttons itself are controled by a custom-element. I hope when we merge it into the release branch the buttons will appear like before?

**Comment by roland-d** [comment]
@coolcat-creations I don't think so because @euismod2336 has already merged the release branch into his branch.

**Comment by coolcat-creations** [comment]
Ok maybe @bembelimen can help on this or knows someone who can help - because the toolbar is a book with seven seals for me.

**Comment by roland-d** [comment]
@coolcat-creations It is nothing to do with the custom-elements. The CSS needs to be updated because the buttons are no longer under the .subhead class.

Or the markup has to be changed to match the CSS. One of the two will do it.

**Comment by euismod2336** [comment]
Well, i can add the extra dom elements to make it match, but that would clutter the dom with useless elements

**Comment by coolcat-creations** [comment]
I probably fixed already the css for the buttons - I just don't know how to test this now. Can you pull release into yours to check the buttons itself? Otherwise: can ist be merged into "release" @bembelimen ?

**Comment by coolcat-creations** [comment]
Some things if possible (if not, someone else needs to do this in the main repo)
- make the Filter Information Bubble only appear if <= 1 Item filtered
- change dropdown to the right side
- Clear Button should clear Search too
- Change css Markup of clear button to btn btn-danger 
- Change css Markup of help and options to btn btn-link
- the buttons in the first line, eg actions, new,... should all have additionally the class btn-sm

**Comment by euismod2336** [comment]
@coolcat-creations I've modified the HTML to contain the classes. Only thing I can't do is displaying the dropdown to the right, as this is all CSS (for the JS it doesn't matter where you click on the button). If you or someone else could look into that, would be great! :)

**Comment by coolcat-creations** [comment]
Thanks a lot :) Hope someone can merge this then very soon so I fix the remaining issues :)

**Comment by euismod2336** [comment]
there is the merge button ;)

**Comment by euismod2336** [comment]
@bembelimen can this be merged?

**Comment by bembelimen** [comment]
Hello @euismod2336 thanks for your PR, I would like to have the [breadcrums ](https://github.com/joomla/backend-template/pull/187) done and then we can merge your PR.

**Comment by euismod2336** [comment]
@bembelimen let me know when it makes sense to resolve the conflicts

**Comment by Hackwar** [comment]
I guess that would be now then...

**Comment by brianteeman** [comment]
When you are resolving the conflicts can you also please fix the codestyle. I am specifically referring to the tab indentation which is all over the place

**Comment by bembelimen** [comment]
As breadcrumbs are gone and the settings/help buttons are normal again (due to a11y) I think this PR makes not much sense anymore.

Thanks for your work.

**Pull request closed without merging**